index.js

```diff
@@ -223,14 +222,15 @@ function create (opts) {
     opened[hex] = old
 
     if (!feed) {
-      old.feed = feed = core.createFeed(key, {
-        storage: storage(path.join(dir, 'data', key.slice(0, 2), key.slice(2) + '.data')),
-        sparse: opts.sparse && isContent
-      })
+      old.feed = feed = hypercore(
+        path.join(dir, 'data', key.slice(0, 2), key.slice(2) + '.data'),
+        key,
+        {sparse: opts.sparse && isContent}
+      )
     }
 
     old.cnt++
-    feed.replicate({stream: stream})
+    var replicator = feed.replicate({stream: stream})
```

Comment:
@mafintosh not sure if my logic is right here on replication, seems like replicator.on('end' is never firing